source/app.d

```diff
@@ -433,33 +434,28 @@ void dedupTravisBuilds(string action, string repoSlug)
     }
 
     auto url = "https://api.travis-ci.org/repos/%s/builds?event_type=pull_request".format(repoSlug);
-    auto activeBuilds = requestHTTP(url, (scope req) {
+    auto activeBuildsForPR = requestHTTP(url, (scope req) {
             req.headers["Authorization"] = travisAuth;
             req.headers["Accept"] = "application/vnd.travis-ci.2+json";
         })
         .readJson["builds"][]
-        .filter!(b => activeState(b["state"].get!string));
-    // builds are sorted from new to old, and also paginated by 25
-    // (but we'll hardly have 25 active builds at the same time)
-    bool[uint] seen;
-    foreach (b; activeBuilds)
-    {
-        immutable pr = b["pull_request_number"].get!uint;
-        if (pr in seen)
-            cancelBuild(b["id"].get!size_t);
-        else
-            seen[pr] = true;
-    }
+        .filter!(b => activeState(b["state"].get!string))
+        .filter!(b => b["pull_request_number"].get!uint == pullRequestNumber);
+
+    // Keep only the most recent build for this PR.  Kill all builds
+    // when it got merged as it'll be retested after the merge anyhow.
+    foreach (b; activeBuildsForPR.drop(action == "merged" ? 0 : 1))
```

Comment:
well ideally we only merge if travis passes, but that's a [different issue](https://github.com/braddr/d-tester/issues/52) ...